code/game/objects/items/weapons/twohanded.dm

```diff
@@ -405,6 +406,32 @@
 	if(wielded)
 		. = ..()
 
+
+/obj/item/weapon/twohanded/required/chainsaw/doomslayer
+	name = "OOOH BABY"
+	desc = "<span class='warn'>VRRRRRRR!!!</span>"
+	force = 20
+	armour_penetration = 100
+
+/obj/item/weapon/twohanded/required/chainsaw/doomslayer/attack_self(mob/user)
+	on = !on
+	user << "You viciously yank the cord dangling from \the [src], [on ? "and it roars to life, screaming along with your bloodlust." : "grows dormant."]"
+	force = on ? 80 : 20
```

Comment:
I think we have a variable for like "on force", but if not make one here for the chainsaw (to again help remove this proc's copypasta)